src/app/clusters/service-area-server/service-area-cluster-objects.h

```diff
@@ -0,0 +1,368 @@
+             const DataModel::Nullable<int16_t> & aFloorNumber, const DataModel::Nullable<AreaTypeTag> & aAreaType,
+             const DataModel::Nullable<LandmarkTag> & aLandmarkTag, const DataModel::Nullable<PositionTag> & aPositionTag,
+             const DataModel::Nullable<FloorSurfaceTag> & aSurfaceTag)
+    {
+        locationID = aLocationId;
+        mapID      = aMapId;
+
+        // If there is at least one non-null value for locationInfo, add it to the location structure.
+        if ((!aLocationName.empty()) || (!aFloorNumber.IsNull()) || (!aAreaType.IsNull()))
+        {
+            // Create a home location info structure and fill it in except for the location name. This is done below.
+            locationInfo.locationInfo.SetNonNull(Structs::HomeLocationStruct::Type());
+
+            locationInfo.locationInfo.Value().floorNumber = aFloorNumber;
+            locationInfo.locationInfo.Value().areaType    = aAreaType;
+        }
+        else
+        {
+            locationInfo.locationInfo.SetNull();
+        }
+
+        locationInfo.landmarkTag = aLandmarkTag;
+        locationInfo.positionTag = aPositionTag;
+        locationInfo.surfaceTag  = aSurfaceTag;
+
+        // Save the location name to buffer, so access and lifetime is controlled.
+        // If aLocationName is not null, create a CharSpan to reference the text.
+        // Make sure there is always a terminating character, mainly so we can get c_str for logging.
+        memset(mLocationNameBuffer, 0, sizeof(mLocationNameBuffer + 1));
+
+        // this assumes locationInfo structure was created above, if appropriate
+        if (!locationInfo.locationInfo.IsNull())
+        {
+            if (aLocationName.empty())
+            {
+                locationInfo.locationInfo.Value().locationName = CharSpan(mLocationNameBuffer, 0);
+            }
+            else if (aLocationName.size() > sizeof(mLocationNameBuffer))
+            {
+                // Save the truncated name that fits into available size.
+                memcpy(mLocationNameBuffer, aLocationName.data(), sizeof(mLocationNameBuffer));
+                locationInfo.locationInfo.Value().locationName = CharSpan(mLocationNameBuffer, sizeof(mLocationNameBuffer));
+            }
+            else
+            {
+                // Save full name.
+                memcpy(mLocationNameBuffer, aLocationName.data(), aLocationName.size());
+                locationInfo.locationInfo.Value().locationName = CharSpan(mLocationNameBuffer, aLocationName.size());
+            }
+        }
+    }
+
+    /**
+     * @brief Compare the location's name with the given text.
+     * @param[in] aLocationName The name to compare.
+     * @return true if the location structure's name field matches aLocationName.
+     *
+     * @note if locations structure's name field is null, returns false.
+     */
+    bool DoesNameMatch(const CharSpan & aLocationName) const
```

Comment:
Why did you call somethingg that calls DataEqual be `namematch` ?
Could we name it `LocationNameDataEqual` or similar?

Reply:
This has been renamed to `IsNameEqual` [here](https://github.com/project-chip/connectedhomeip/pull/33991/commits/c1b9e3164a15be61beda1a04543166b05b574763).